components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/AnalyzerImpl.java

```diff
@@ -0,0 +1,149 @@
+/*
+* Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.wso2.carbon.apimgt.core.impl;
+
+import org.wso2.carbon.apimgt.core.api.Analyzer;
+import org.wso2.carbon.apimgt.core.dao.AnalyticsDAO;
+import org.wso2.carbon.apimgt.core.exception.APIManagementException;
+import org.wso2.carbon.apimgt.core.exception.APIMgtDAOException;
+import org.wso2.carbon.apimgt.core.exception.AnalyticsException;
+import org.wso2.carbon.apimgt.core.exception.ExceptionCodes;
+import org.wso2.carbon.apimgt.core.models.analytics.APICount;
+import org.wso2.carbon.apimgt.core.models.analytics.APIInfo;
+import org.wso2.carbon.apimgt.core.models.analytics.APISubscriptionCount;
+import org.wso2.carbon.apimgt.core.models.analytics.ApplicationCount;
+import org.wso2.carbon.apimgt.core.models.analytics.SubscriptionCount;
+import org.wso2.carbon.apimgt.core.models.analytics.SubscriptionInfo;
+
+import java.time.Instant;
+import java.util.List;
+
+/**
+ * Implementation class of Analyzer operations.
+ */
+public class AnalyzerImpl implements Analyzer {
+
+    private String username;
+    private AnalyticsDAO analyticsDAO;
+
+    public AnalyzerImpl(String username, AnalyticsDAO analyticsDAO) {
+        this.username = username;
+        this.analyticsDAO = analyticsDAO;
+    }
+
+    /**
+     * @see Analyzer#getApplicationCount(Instant, Instant, String)
```

Comment:
Shall we add some debug logs to all the methods. at least to log the requested  properties and out put